pytorch_lightning/trainer/training_loop.py

```diff
@@ -841,6 +842,9 @@ def training_step_and_backward(self, split_batch, batch_idx, opt_idx, optimizer,
         return result
 
     def backward(self, result, optimizer, opt_idx, *args, **kwargs):
+        if not self._do_backward:
```

Comment:
The only thing I don't like about this is that it's elusive. Why wouldn't we call backwards? How as a user do I know if my code is preventing backwards to be called?

I understand it's an internal function but I can see this being an issue down the line. 

At the very least it should be `self.skip_backwards` with a comment on the lines of `# check to see if backwards should be skipped, useful for callbacks`

Reply:
Removed it, but we lose in speed as gradients will be computed for nothing.